### ISSUE TITLE
Samsung Note 4 Exynos Variant Incompatibility

### ISSUE THREAD

**Issue by GentleBrine** [depressed-pho/natural-mystic-shaders#50]
The shaders looks nice in itself. But it doesn't work on my Samsung Note 4 with Mali T720 and Exynos 5433.

**Comment by MrAgentBlaze** [comment]
True, even i had to test it on Android (by chatting to PHO on Discord) and the beta still doesn't work

**Comment by depressed-pho** [comment]
So there is something special in Mali-T720 and Mali-T628. And possibly other variants of Mali. Both of those GPUs allegedly support OpenGL ES 3.1 but I suspect there is a bug in the driver used for those chips. Or perhaps they have exceptionally small limits for varying variables or the number of instructions?

**Comment by depressed-pho** [comment]
Submitted a feature request to debug this issue: [Bedrock should display errors when a shader program fails to compile or link – Minecraft Feedback](https://feedback.minecraft.net/hc/en-us/community/posts/360039424351-Bedrock-should-display-errors-when-a-shader-program-fails-to-compile-or-link)
It is virtually impossible to fix it until they implement this feature.

**Comment by depressed-pho** [comment]
The feature request was deleted FOR SOME REASON. So I submitted it to the bug tracker instead. Let's see how it goes. https://bugs.mojang.com/browse/MCPE-41076

**Comment by depressed-pho** [comment]
Affected GPUs:
* Mali-T628
* Mali-T720
* Mali-G76MP10
* Adreno 306
